var/spack/repos/spack_repo/builtin/packages/scorep/package.py

```diff
@@ -86,29 +86,51 @@ def url_for_version(self, version):
         sha256="d20b3046ba6a89ad9c106bcf372bceb1bd9ab780d4c7dd9e7373f0099b92d933",
     )
 
+    # Variants
     variant("mpi", default=True, description="Enable MPI support")
     variant("papi", default=True, description="Enable PAPI")
-    variant("pdt", default=False, description="Enable PDT")
+    variant("pdt", default=False, description="Enable PDT", when="@:8.4")
     variant("shmem", default=False, description="Enable shmem tracing")
     variant("unwind", default=False, description="Enable sampling via libunwind and lib wrapping")
     variant("cuda", default=False, description="Enable CUDA support")
     variant("hip", default=False, description="Enable ROCm/HIP support", when="@8.0:")
     variant("gcc-plugin", default=True, description="Enable gcc-plugin", when="%gcc")
+    variant(
+        "llvm-plugin", default=True, description="Enable LLVM compiler plugin", when="@9.0: ^llvm"
+    )
+    variant(
+        "binutils",
+        default=True,
+        description="Enable debug info lookup via binutils",
+        when="^binutils",
+    )
+    # Putting this in as preparation. F08 support exists in 9.0 but configure does not respect
+    # --enable-mpi-f08 and will not until 9.1.
+    variant("mpi_f08", default=True, description="Enable MPI F08 support", when="@9.1: +mpi")
     # Dependencies for SCORE-P are quite tight. See the homepage for more
     # information. Starting with scorep 4.0 / cube 4.4, Score-P only depends on
     # two components of cube -- cubew and cubelib.
 
+    # Language dependencies
+    # TODO: we could allow a +fortran variant here.
     depends_on("c", type="build")  # generated
     depends_on("cxx", type="build")  # generated
     depends_on("fortran", type="build")  # generated
 
+    # SCOREP 9
+    depends_on("gotcha@1.0.8:", type="link", when="@9:")
+    depends_on("otf2@3.1:", when="@9:")
+    depends_on("cubew@4.9:", when="@9:")
+    depends_on("cubelib@4.9:", when="@9:")
+    depends_on("opari2@2.0.9", when="@9:")
+
     # SCOREP 8
     depends_on("binutils", type="link", when="@8:")
     depends_on("otf2@3:", when="@8:")
-    depends_on("cubew@4.8.2:", when="@8.3:")
-    depends_on("cubelib@4.8.2:", when="@8.3:")
-    depends_on("cubew@4.8:", when="@8:8.2")
-    depends_on("cubelib@4.8:", when="@8:8.2")
+    depends_on("cubew@4.8.2:4.8.99", when="@8.3:")
+    depends_on("cubelib@4.8.2:4.8.99", when="@8.3:")
+    depends_on("cubew@4.8:4.8.99", when="@8:8.2")
+    depends_on("cubelib@4.8:4.8.99", when="@8:8.2")
```

Comment:
These are equivalent to what I think you want.

```suggestion
    depends_on("cubew@4.8.2:4.8", when="@8.3:")
    depends_on("cubelib@4.8.2:4.8", when="@8.3:")
    depends_on("cubew@4.8", when="@8:8.2")
    depends_on("cubelib@4.8", when="@8:8.2")
```